src/public/modules/forms/helpers/response-factory.ts

```diff
@@ -0,0 +1,65 @@
+import { hasProp } from '../../../../shared/util/has-prop'
+import {
+  ArrayResponse,
+  DisplayedResponseWithoutAnswer,
+  NestedResponse,
+  SingleResponse,
+} from '../../../../types/response'
+
+import {
+  ArrayAnswerResponse,
+  Response,
+  SingleAnswerResponse,
+  TableResponse,
+} from './csv-response-classes'
+
+export const getResponseInstance = (
```

Comment:
I did think about how to add more clarity to the type signature so that it would be clear that an error could be created here. But the benefit of the current approach is that the error can just bubble up to `submissions.client.factory` and it doesn't have to be handled directly in `addRecord`. I did add the comments to the relevant portions of the code to indicate the possibility of an error. Do you think this is sufficient?

Reply:
i think not handling here is ok but the comments have to be in the jsdoc, because otherwise they won't be reflected on mouse hover @__@ 

the main problem is that errors if not stated in the signature, should be plainly obvious to the caller! adding the comment is good but cos it's not in the JSDoc, callers won't see on hover, which defeats the purpose of the comment

Reply:
Oh right... I'll add it in thank you!!